### PR TITLE
Add new endpoint to vaos and mobile module for VPG appointment slots

### PR DESCRIPTION
**Pull request by devin-mccurdy**
## Summary

- This work is behind a feature toggle (flipper): YES  - `va_online_scheduling_enable_OH_slots_search`
- Added a new endpoint to the vaos and mobile modules to get available appointment slots facility slots from the new VA Mobile VPG endpoint, which handles both VISTA and Oracle Health facilities.
- A new endpoint was needed in order to remove the `clinic_id` from the path because it is a field that doesn't exist for Oracle Health facilities. For the new endpoint, either the `clinic_id`(for a Vista facility) or the `clinical_service` (for an Oracle Health facility) fields are required.
- Team: Unified Appointment Experience team

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/76083

## Testing done

- [x] *New code is covered by unit tests*
- The available appointment slots from VISTA facilities are returned from the VAOS endpoint.
- Using the current vets-api slots endpoint, slots are returned correctly for facilities. Future vets-website changes will be required to make use of the new endpoint in order to also retrieve available slots for Oracle Health facilities

## What areas of the site does it impact?
This affects the vaos and mobile apps. When the feature flag is ON, flags are returned from the va-mobile / MAP vaos endpoint instead of the vaos endpoint, but the slot data structures and source systems are the same for Vista facilities and all existing functionality is supported.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by drewconnelly — APPROVED** [review]
LGTM

File: modules/mobile/app/controllers/mobile/v0/clinics_controller.rb [hunk 1]
```diff
@@ -16,12 +16,37 @@ def slots
...
+                                                       clinical_service: nil,
                                                        start_dt: start_date,
                                                        end_dt: end_date)
 
         render json: Mobile::V0::ClinicSlotsSerializer.new(response)
       end
 
+      def facility_slots
+        if !params[:clinic_id] && !params[:clinical_service]
+          render status: :bad_request, json: {
```
Comment (LindseySaari) [inline modules/mobile/app/controllers/mobile/v0/clinics_controller.rb:28]: for readbility, `unless params[:clinic_id] || params[:clinical_service]` to me makes it clearer that you're checking for the absence of both parameters

Comment (devin-mccurdy) [reply modules/mobile/app/controllers/mobile/v0/clinics_controller.rb:28]: Rubocop didn't like an `unless` with an `else`, and went with `if params[:clinic_id] || params[:clinical_service]`.  Is that better?

**Review by drewconnelly — APPROVED** [review]
Approved.

**Review by stevenjcumming — APPROVED** [review]

**Review by cferris32 — APPROVED** [review]
LGTM for the VAOS module

**Comment by devin-mccurdy** [comment]
@jperk51 could I get a `mobile` code owner to review this? Thanks!

**Review by jperk51 — APPROVED** [review]
LGTM

**Pull request merged**